skpro/regression/cyclic_boosting.py

```diff
@@ -0,0 +1,315 @@
+    >>> X, y = load_diabetes(return_X_y=True, as_frame=True)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y)
+    >>>
+    >>> fp = {
+    >>>     'age': flags.IS_CONTINUOUS,
+    >>>     'sex': flags.IS_CONTINUOUS,
+    >>>     'bmi': flags.IS_CONTINUOUS,
+    >>>     'bp':  flags.IS_CONTINUOUS,
+    >>>     's1':  flags.IS_CONTINUOUS,
+    >>>     's2':  flags.IS_CONTINUOUS,
+    >>>     's3':  flags.IS_CONTINUOUS,
+    >>>     's4':  flags.IS_CONTINUOUS,
+    >>>     's5':  flags.IS_CONTINUOUS,
+    >>>     's6':  flags.IS_CONTINUOUS,
+    >>> }
+    >>> reg_proba = CyclicBoosting(feature_properties=fp)
+    >>> reg_proba.fit(X_train, y_train)
+    >>> y_pred = reg_proba.predict_proba(X_test)
+    """
+
+    _tags = {
+        "object_type": "regressor",
+        "estimator_type": "regressor_proba",
+        "capability:multioutput": False,
+        "capability:missing": True,
+        "X_inner_mtype": "pd_DataFrame_Table",
+        "y_inner_mtype": "pd_DataFrame_Table",
+        "python_dependencies": "cyclic_boosting>=1.2.1",
+    }
+
+    def __init__(self, feature_properties, interaction=tuple(), distr_type="Normal"):
+        self.feature_properties = feature_properties
+        self.interaction = interaction
+        self.quantiles = [0.2, 0.5, 0.8]
+        self.quantile_values = []
+        self.quantile_est = []
+        self.qpd = None
+        self.distr_type = distr_type
+
+        super().__init__()
+
+        # check parameters
+        if not isinstance(feature_properties, dict):
+            raise ValueError("feature_properties must be dict")
+        for i in interaction:
+            if not isinstance(i, tuple):
+                raise ValueError("interaction must be tuple")
+
+        # build estimators
+        features = list(self.feature_properties.keys())
+        for i in interaction:
+            features.append(i)
+
+        for quantile in self.quantiles:
+            self.quantile_est.append(
+                pipeline_CBMultiplicativeQuantileRegressor(
+                    quantile=quantile,
+                    feature_properties=self.feature_properties,
+                    feature_groups=features,
+                    maximal_iterations=50,
+                )
+            )
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+
+        self._y_cols = y.columns
+        y = y.to_numpy().flatten()
+
+        # multiple quantile regression for full probability estimation
+        for est in self.quantile_est:
+            est.fit(X.copy(), y)
+
+        return self
+
+    def _predict(self, X):
```

Comment:
I would suggest to not run the full quantile regression - QPD thing here, but simply use a mean prediction mode (e.g., pipeline_CBPoissonRegressor for a multiplicative model on range 0, inf).

Reply:
But that would require two different fit modes (as we want to use quantile regression with subsequent QPD for predict_proba), from which the user would have to select via a parameter in the init of the estimator. So, I can understand your rationale of using the mean of QPD. If we wanted to keep a single fit mode, we might consider using just the median (anyway predicted by the middle quantile regression model for QPD) though (avoiding the complexity via the mean of QPD).

Reply:
@FelixWick 
How about Having two kind of estimators in Cyclic boosting class? I think it look seemless from user because keeping the single fit mode. Is it bit complicated?

Reply:
Yes, that is what I meant. But at fit time, the user needs to decide what estimator to train. It would be a waste of runtime to train both always. (That is also true for the quantile regression of three different quantiles for the QPD, if all you want is the median.)

Reply:
is it perhaps helpful to look at the QR model by @Ram0nB here?
https://github.com/sktime/skpro/blob/main/skpro/regression/multiquantile.py

Reply:
Ok, I changed my mind. For simplicity, what do you think of the following structure:
- always fit three quantiles: alpha, 0.5 (median), and 1-alpha (alpha to be defined as parameter instead of quantiles triplet). That means just use quantile regression as single estimator. But we should use either multiplicative or additive mode, depending on the target range.
- in predict_proba, give QPD as distribution.
- in predict, give the median as point estimate.
- in predict_quantiles, give the three quantile estimates (alpha, median, 1-alpha), or, if that's not possible from the interface, just the alpha estimate.

Reply:
@FelixWick 
I followed your comments. Additionally, predict_interval method was implemented. In predict_quantiles, if alphas are same with the quantiles that the estimator train, the method returns estimated value directlly,  if it is not same, the method returns specified quantile value from QPD.